scripts/unity_dataset_processing/hssd_to_unity_data.py

```diff
@@ -0,0 +1,352 @@
+                for model_id in model_ids:
+                    for render_glb in render_glbs:
+                        if model_id + ".glb" in render_glb:
+                            if "part" in render_glb and "part" not in model_id:
+                                continue
+                            model_filepaths.add(render_glb)
+
+    return model_filepaths
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Get all .glb render asset files associated with a given scene."
+    )
+    parser.add_argument(
+        "--hssd-hab-root-dir",
+        type=str,
+        help="Path to the hssd-hab root directory containing 'hssd-hab-uncluttered.scene_dataset_config.json'.",
+    )
+    parser.add_argument(
+        "--hssd-models-root-dir",
+        type=str,
+        help="Path to hssd-models root directory.",
+    )
+    parser.add_argument(
+        "--scenes",
+        nargs="+",
+        type=str,
+        help="one or more scene ids",
+    )
+
+    args = parser.parse_args()
+
+    # Force input paths to have a trailing slash
+    if args.hssd_hab_root_dir[-1] != "/":
+        args.hssd_hab_root_dir += "/"
+    if args.hssd_models_root_dir[-1] != "/":
+        args.hssd_models_root_dir += "/"
+
+    jobs: List[Job] = []
+    scene_ids = list(dict.fromkeys(args.scenes)) if args.scenes else []
+
+    # Define relative directory for the dataset
+    # E.g. data/scene_datasets/hssd-hab -> scene_datasets/hssd-hab
+    hssd_hab_rel_dir = str(args.hssd_hab_root_dir)[len("data/") :]
+
+    # Add stages
+    for scene_id in scene_ids:
+        rel_path = os.path.join("stages", scene_id + ".glb")
+        job = Job()
+        job.source_path = os.path.join(args.hssd_hab_root_dir, rel_path)
+        job.dest_path = os.path.join(OUTPUT_DIR, hssd_hab_rel_dir, rel_path)
+        job.simplify = False
+        jobs.append(job)
+
+    # Add all models contained in the scenes
+    scene_models = find_model_paths_in_scenes(args.hssd_hab_root_dir, scene_ids)
+    for scene_model in scene_models:
+        rel_path = scene_model[len(args.hssd_hab_root_dir) :]
+        if "decomposed" not in scene_model:
+            job = Job()
+            source_path = os.path.join(args.hssd_models_root_dir, rel_path)
+            parts = source_path.split(
+                "/objects/"
+            )  # Remove 'objects/' from path
+            job.source_path = os.path.join(parts[0], parts[1])
+            assert len(parts) == 2
+            job.dest_path = os.path.join(OUTPUT_DIR, hssd_hab_rel_dir, rel_path)
+            job.simplify = False
+            jobs.append(job)
+        else:
+            job = Job()
+            job.source_path = os.path.join(args.hssd_hab_root_dir, rel_path)
+            job.dest_path = os.path.join(OUTPUT_DIR, hssd_hab_rel_dir, rel_path)
+            job.simplify = False
+            jobs.append(job)
+
+    # Add ycb objects
+    for filename in Path("data/objects/ycb/meshes").rglob("*.glb"):
```

Comment:
This script is doing stuff with ycb and Spot assets, so I suggest avoid using `hssd_to_unity_data.py` as the name. It is not just for HSSD.